### ISSUE TITLE
monerod: Failed to delete transaction prunable data

### ISSUE THREAD

**Issue by ukfhVp0zms** [monero-project/monero#5644]
Receiving an error when trying to prune blockchain data

Manjaro Linux

`~ >>> monerod --hide-my-port --prune-blockchain`
`2019-06-14 23:06:37.651 I Monero 'Boron Butterfly' (v0.14.1.0-release)`
`2019-06-14 23:06:37.651 I Initializing cryptonote protocol...`
`2019-06-14 23:06:37.651 I Cryptonote protocol initialized OK`
`2019-06-14 23:06:37.651 I Initializing p2p server...`
`2019-06-14 23:06:43.129 I p2p server initialized OK`
`2019-06-14 23:06:43.130 I Initializing core RPC server...`
`2019-06-14 23:06:43.130 I Binding on 127.0.0.1:18081`
`2019-06-14 23:06:43.141 I Generating SSL certificate`
`2019-06-14 23:06:43.243 I core RPC server initialized OK on port: 18081`
`2019-06-14 23:06:43.243 I Initializing core...`
`2019-06-14 23:06:43.243 I Loading blockchain from folder /home/xxx/.bitmonero/lmdb ...`
`2019-06-14 23:06:43.289 I Loading checkpoints`
`2019-06-14 23:06:43.653 I Pruning blockchain...`
`2019-06-14 23:08:04.046 W Failed to delete transaction prunable data: MDB_TXN_FULL: Transaction has too many dirty pages - transaction too big`
`2019-06-14 23:08:04.055 F Uncaught exception! Failed to delete transaction prunable data: MDB_TXN_FULL: Transaction has too many dirty pages - transaction too big`
`2019-06-14 23:08:04.105 I Deinitializing core RPC server...`
`2019-06-14 23:08:04.107 I Deinitializing p2p...`
`2019-06-14 23:08:08.111 I Deinitializing core...`
`2019-06-14 23:08:09.269 I Stopping cryptonote protocol...`
`2019-06-14 23:08:09.269 I Cryptonote protocol stopped successfully`

**Comment by moneromooo-monero** [comment]
I'll need to have checkpoints to commit db txns along the way. In the meantime, you can use monero-blockchain-prune, which should not have this problem.

**Comment by moneromooo-monero** [comment]
https://github.com/monero-project/monero/pull/5647 should also fix it.

**Comment by BKdilse** [comment]
I used monero-blockchain-prune to prune my DB, and it went through fine, however, once I restarted the daemon, I got some errors to do with tx/block not in DB (I didn't get a chance to note this, sorry), priority was to get my pool back up, as it would not sync past the last block it was on, even waited 15 mins.  I am currently re-synching from scratch, using --prune-blockchain.

**Comment by moneromooo-monero** [comment]
If it was "Error retrieving blocks, missed X transactions for block with hash Y", then it's fine, it's old nodes not knowing you don't have that data and asking for it anyway.

**Comment by BKdilse** [comment]
No, it wasn't that, sorry, i'm not being much help by not grabbing the logs, it was early in the morning, I was tired, and just wanted pool fixed :(

I was hoping someone else has experienced it, and had grabbed the error details.

**Comment by dwjorgeb** [comment]
@moneromooo-monero I just had this same error, when pruning with the flag. 

Then pruned with the tool and everything seemed fine, but when rebooted the node it got stuck. 

No errors besides ```Error retrieving blocks, missed 14 transactions for block with hash: <4f058f1674bc7114f30745d06c292134f52c2fbcbe30d0048080946942b7503f>``` once, but it's stuck for 20 mins on the same block and won't sync.

Going to resync from scratch now

**Comment by moneromooo-monero** [comment]
If it's stuck and won't sync, restart with --log-level 1 to see why it won't sync.

**Comment by moneromooo-monero** [comment]
ping

That last message is normal, it's from nodes that don't know about pruning yet and asking you for data you don't have. It's silenced in current code. It should not wedge your node at all though. 
Run with --log-level 1, and get a stack trace (gdb PATHTOMONEROD PIDOFMONEROD, thread apply all bt).
